tests/integration/helm_fluentbit_containerd_multiline_logs_test.go

```diff
@@ -0,0 +1,112 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal"
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/ctxopts"
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/stepfuncs"
+)
+
+func Test_Helm_FluentBit_Containerd_Multiline_Logs(t *testing.T) {
+	const (
+		tickDuration           = 3 * time.Second
+		waitDuration           = 3 * time.Minute
+		logRecords             = 4
+		logLoops               = 500
+		multilineLogCount uint = logRecords * logLoops
```

Comment:
Maybe add some comments that these values are hardcoded in the pod definition yaml?

Or maybe even better: instead of applying a `.yaml` that's saved on disk we could define the pod in code and then we could have control over the loop counts and we wouldn't have to rely on "implicit dependency" between these `consts` here and the pod `.yaml` file?

WDYT?

Reply:
As a point of reference, we've defined a similar thing in code here: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/tests/integration/internal/logsgenerator/logsgenerator.go

Reply:
I've created an issue for this so that we can perhaps make this use less hardcoded and (very) loosely coupled `.yaml` files.

https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/2055

Ideally we could use that approach elsewhere as well, e.g. in the place where you mentioned it @swiatekm-sumo

Reply:
It will be good to have everything in the code but unfortunately size of arguments for processes is limited, related articles [1](https://www.in-ulm.de/~mascheck/various/argmax/), [2](https://www.baeldung.com/linux/argument-list-too-long-error) 

Following error occurred:
```
cat /var/log/containers/example-pod-multiline-logs-long-lines_multiline-logs-generator_example-pod-multiline-logs-long-lines-3cd7f6f3fd6f2a4eda236a9c55882beb05efe78f3e18e2695b09d7b635489433.log 
2022-02-22T13:57:55.662403778Z stderr F standard_init_linux.go:228: exec user process caused: argument list too long
```
and because of that we cannot pass explicitly very long strings  as arguments to `echo` and we cannot create very long number of arguments as `args` in Pod specification

Experiments with changing Pod in `yaml` to `go` are available on branch: https://github.com/SumoLogic/sumologic-kubernetes-collection/tree/kk-test-multiline-logs-trial-to-replace-yaml-with-go

Reply:
Thanks for checking if it's possible 🙏 

Ideally I'd like to have that documented somewhere, perhaps an issue? Perhaps a comment? So that we know we've investigated that and we know what are the limitations.

Maybe a "known limitations/issues" section in tests readme?